### PR TITLE
Removed note on concurrency group/matrix usage

### PR DESCRIPTION
**Pull request by james2791**
The note below in the `build_matrix` page indicates that you are able to use matrix `setup` dimension values inside concurrency groups (i.e name): like the below

```
steps:
- label: "A Matrix Build"
  command: "echo {{matrix.os}} {{matrix.arch}} {{matrix.test}}"
  concurrency: 1
  concurrency_group: "group-{{matrix.arch}}"
  matrix:
    setup:
      arch:
        - "amd64"
      os:
        - "macos"
      test:
        - "alpha"
```
That interpolation for the CG's name at the moment isn't supported - and will cause confusion for those hoping the dynamic naming/support can work in current form (either hard coded names/plugins are the way around this)

### CODE REVIEW

File: pages/pipelines/build_matrix.md [hunk 1]
```diff
@@ -11,9 +11,6 @@ The following [command step](/docs/pipelines/command-step) attributes can contai
 * [plugins](/docs/pipelines/command-step#plugins)
 * [agents](/docs/pipelines/command-step#agents)
 
-You can't use matrix values in other attributes, including step keys, but you can use them in [concurrency groups](/docs/pipelines/controlling-concurrency#concurrency-groups).
-
-
```
Comment (alexframe) [inline pages/pipelines/build_matrix.md:16]:
```suggestion
You can't use matrix values in other attributes, including step keys and concurrency groups.
```

Comment (james2791) [reply pages/pipelines/build_matrix.md]: As suggested @alexframe 💚

File: pages/pipelines/build_matrix.md [hunk 2]
```diff
@@ -11,8 +11,7 @@ The following [command step](/docs/pipelines/command-step) attributes can contai
 * [plugins](/docs/pipelines/command-step#plugins)
 * [agents](/docs/pipelines/command-step#agents)
 
-You can't use matrix values in other attributes, including step keys, but you can use them in [concurrency groups](/docs/pipelines/controlling-concurrency#concurrency-groups).
-
+You can't use matrix values in other attributes, including step keys and concurrency groups.
```
Comment (mbelton-buildkite) [inline pages/pipelines/build_matrix.md:14]: Was removing the link intentional?

I think it's probably best to keep it if not:

```suggestion
You can't use matrix values in other attributes, including step keys and [concurrency groups](/docs/pipelines/controlling-concurrency#concurrency-groups).
```

**Review by mbelton-buildkite — APPROVED** [review]
One minor nitpick, then this looks all good :)

File: pages/pipelines/build_matrix.md (hunk 2 above)
Comment (james2791) [reply pages/pipelines/build_matrix.md]: Most likely oversight - I added it back @mbelton-buildkite 💚

**Pull request merged**